app/src/screens/prove/QRCodeViewFinderScreen.tsx

```diff
@@ -88,8 +92,10 @@ const QRCodeViewFinderScreen: React.FC = () => {
           trackEvent(ProofEvents.QR_SCAN_SUCCESS, {
             scan_type: 'sessionId',
           });
-          useSelfAppStore.getState().cleanSelfApp();
-          useSelfAppStore.getState().startAppListener(sessionId);
+
+          selfClient.getSelfAppState().cleanSelfApp();
```

Comment:
do we call `getSelfAppState()` so that we get the most up to date state?

could we do something like:

```
const selfState = useSelfAppStore.getSelfAppState();
```